.github/workflows/pr-assess-bundle-size.yml

```diff
@@ -0,0 +1,42 @@
+name: Compressed Size
+
+on:
+    pull_request:
+        paths:
+            - '**.jsx?'
+            - '**.tsx?'
+            - '**.css'
+            - '**.scss'
+            - '**package*.json'
+            - '**.eslint*'
+            - '**.prettier*'
+            - '**.tsconfig*'
+            - '**/webpack.config.js'
+            - '.github/workflows/pr-assess-bundle-size.yml'
+
+concurrency:
+    group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
+    cancel-in-progress: true
+
+jobs:
+    build:
+        name: Check Asset Sizes
+        runs-on: ubuntu-latest
+        permissions:
+            pull-requests: write
+
+        steps:
+            - uses: actions/checkout@v4
+            - name: Setup WooCommerce Monorepo
+              uses: ./.github/actions/setup-woocommerce-monorepo
+              with:
+                  # Both install and build are handled by compressed-size-action.
+                  install: false
+                  build: false
+            - uses: preactjs/compressed-size-action@f780fd104362cfce9e118f9198df2ee37d12946c
+              with:
+                  repo-token: '${{ secrets.GITHUB_TOKEN }}'
+                  pattern: './{packages/js/!(*e2e*|*internal*|*test*|*plugin*|*create*),plugins/woocommerce-blocks}/{build,build-style}/**/*.{js,css}'
+                  clean-script: '--if-present distclean'
```

Comment:
`--if-present` can be removed once this is merged to trunk, it's just trunk doesn't have the command yet and so the comparison couldn't happen. I thought it would be nice to just inline this as a bash script but nope, can't do it with this plugin, it always runs your package manager first. This plugin is really limited actually. I would like to write something more flexible (and maybe better suited to grouping size metrics per package) in future but for now it still gives the features we need.